### ISSUE TITLE
this can be potentialy asynchronous... will asynchronous version be created

### ISSUE THREAD

**Issue by 07pepa** [billdeitrick/pypco#49]

**Comment by billdeitrick** [comment]
An async version would be fantastic. I don't have sufficient margin to do this immediately, but would happily accept a PR as per my comments in #48.

**Comment by 07pepa** [comment]
yeah but i do not know if synchronous version should stay or should be converted

othervise i will likely use this ( it support trio any io and asyncio (curio unfortunately not) 

https://www.python-httpx.org/async/

**Comment by 07pepa** [comment]
also i see it like this types  and testing them even internaly (bigest pain)  

then maybe optional converting to clases 

then async

**Comment by billdeitrick** [comment]
We need to keep the synchronous calls in place to avoid breaking backwards compatibility. My preference with adding async support would be for the synchronous calls to simply become wrappers to async versions of the same calls, and then existing code will continue to work with the option to use async calls as desired.

I think I would like to see async support added before object serialization/deserialization, but we can be flexible there.

I haven't used httpx before, but from a quick read it looks like that would be a good choice. I like the API feels similar to requests and it also has VCRPY support (I would consider this a hard requirement).

**Comment by 07pepa** [comment]
ok i am fine with async going 1st as well 

i think i will do a mirror image of curent api that will be async and wrap it

someting like this 
```python
import asyncio
class PCO:
#... store async version in self and current (created) asyncio loop in self
    def get(self,<PARAMS>):
        return self.loop.run_until_complete(self.async_pco.get(<PARAMS>)) # should be thread safe but documentation does not specify
```

this should be simplest way with no other dependency just httpx  and dropping requests

and current test will stay and will not double and cov still will be 100%

**Comment by 07pepa** [comment]
and there is fix for pyvcr it is currently a blocker https://github.com/kevin1024/vcrpy/pull/657
